servers/rendering/renderer_viewport.cpp

```diff
@@ -145,30 +145,48 @@ void RendererViewport::_configure_3d_render_buffers(Viewport *p_viewport) {
 				scaling_3d_mode = RS::VIEWPORT_SCALING_3D_MODE_OFF;
 			}
 
-			if (scaling_3d_mode == RS::VIEWPORT_SCALING_3D_MODE_METALFX_TEMPORAL && !RD::get_singleton()->has_feature(RD::SUPPORTS_METALFX_TEMPORAL)) {
-				if (RD::get_singleton()->has_feature(RD::SUPPORTS_METALFX_SPATIAL)) {
-					// Prefer MetalFX spatial if it is supported, which will be much more efficient than FSR2,
-					// as the hardware already will struggle with FSR2.
-					scaling_3d_mode = RS::VIEWPORT_SCALING_3D_MODE_METALFX_SPATIAL;
-					WARN_PRINT_ONCE("MetalFX temporal upscaling is not supported by the current renderer or hardware. Falling back to MetalFX Spatial scaling.");
-				} else {
-					scaling_3d_mode = RS::VIEWPORT_SCALING_3D_MODE_FSR2;
-					WARN_PRINT_ONCE("MetalFX upscaling is not supported by the current renderer or hardware. Falling back to FSR 2 scaling.");
+			if (scaling_3d_mode == RS::VIEWPORT_SCALING_3D_MODE_METALFX_TEMPORAL) {
+#ifdef RD_ENABLED
+				if (!RD::get_singleton()->has_feature(RD::SUPPORTS_METALFX_TEMPORAL))
+#endif // RD_ENABLED
+				{
+#ifdef RD_ENABLED
+					if (RD::get_singleton()->has_feature(RD::SUPPORTS_METALFX_SPATIAL)) {
+						// Prefer MetalFX spatial if it is supported, which will be much more efficient than FSR2,
+						// as the hardware already will struggle with FSR2.
+						scaling_3d_mode = RS::VIEWPORT_SCALING_3D_MODE_METALFX_SPATIAL;
+						WARN_PRINT_ONCE("MetalFX temporal upscaling is not supported by the current renderer or hardware. Falling back to MetalFX Spatial scaling.");
+					} else
+#endif // RD_ENABLED
+					{
+						scaling_3d_mode = RS::VIEWPORT_SCALING_3D_MODE_FSR2;
+						WARN_PRINT_ONCE("MetalFX upscaling is not supported by the current renderer or hardware. Falling back to FSR 2 scaling.");
+					}
+					scaling_type = RS::scaling_3d_mode_type(scaling_3d_mode);
 				}
-				scaling_type = RS::scaling_3d_mode_type(scaling_3d_mode);
 			}
 
-			if (scaling_3d_mode == RS::VIEWPORT_SCALING_3D_MODE_METALFX_SPATIAL && !RD::get_singleton()->has_feature(RD::SUPPORTS_METALFX_SPATIAL)) {
-				scaling_3d_mode = RS::VIEWPORT_SCALING_3D_MODE_FSR;
-				WARN_PRINT_ONCE("MetalFX spatial upscaling is not supported by the current renderer or hardware. Falling back to FSR scaling.");
+			if (scaling_3d_mode == RS::VIEWPORT_SCALING_3D_MODE_METALFX_SPATIAL) {
+#ifdef RD_ENABLED
+				if (!RD::get_singleton()->has_feature(RD::SUPPORTS_METALFX_SPATIAL))
+#endif // RD_ENABLED
+				{
+					scaling_3d_mode = RS::VIEWPORT_SCALING_3D_MODE_FSR;
+					WARN_PRINT_ONCE("MetalFX spatial upscaling is not supported by the current renderer or hardware. Falling back to FSR scaling.");
+				}
```

Comment:
This is also pretty convoluted. I think a clear
```
#ifdef RD_ENABLED

// Original logic

#else 

// Fallback to FSR

#endif
```
would be more readable, even if that adds a couple lines of duplication compared to the feature-not-supported case of the `#if` branch.